local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -89,25 +86,25 @@ void WaypointGenerator::calculateWaypoint() {
     }
   }
   last_wp_type_ = planner_info_.waypoint_type;
-  last_position_waypoint_ = output_.position_waypoint;
   last_yaw_ = curr_yaw_;
-  last_velocity_ =
-      Eigen::Vector2f(curr_vel_.twist.linear.x, curr_vel_.twist.linear.y);
 }
 
 void WaypointGenerator::setFOV(float h_FOV, float v_FOV) {
   h_FOV_ = h_FOV;
   v_FOV_ = v_FOV;
 }
 
-void WaypointGenerator::updateState(const geometry_msgs::PoseStamped& act_pose,
-                                    const geometry_msgs::PoseStamped& goal,
-                                    const geometry_msgs::TwistStamped& vel,
+void WaypointGenerator::updateState(const Eigen::Vector3f& act_pose,
+                                    const Eigen::Quaternionf& q,
+                                    const Eigen::Vector3f& goal,
+                                    const Eigen::Vector3f& vel,
                                     bool stay, bool is_airborne) {
-  pose_ = act_pose;
-  curr_vel_ = vel;
-  goal_ = toEigen(goal.pose.position);
-  curr_yaw_ = static_cast<float>(tf::getYaw(pose_.pose.orientation));
+
+  position_ = act_pose;
+  velocity_ = vel;
+  goal_ = goal;
+  Eigen::Vector3f euler = q.toRotationMatrix().eulerAngles(1, 0, 2);
```

Comment:
You can get the yaw inline just using the `.z()` on the output ie. 

`curr_yaw_ = q.toRotationMatrix().eulerAngles(1, 0, 2).z();`